.github/workflows/python-workflow.yml

```diff
@@ -94,3 +98,26 @@ jobs:
         with:
           name: pr
           path: pr/
+
+  # Run mypy on new or modified AGW python files to check typing
+  check-typing:
+    needs: pre_job
+    if: ${{ needs.pre_job.outputs.should_not_skip_agw == 'true' }}
+    runs-on: ubuntu-latest
+    name: AGW Python Typing Check
+    steps:
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8.10'
+      - name: Install mypy
+        run: |
+          pip3 install --upgrade pip
+          pip3 install mypy==0.960
+      - name: Run mypy in AGW
+        run: |
+          for file in ${{ needs.pre_job.outputs.files_changed_agw }};
+          do
+            set -e
+            echo "Running mypy on $file...";
+            mypy --ignore-missing-imports $MAGMA_ROOT/$file;
```

Comment:
We might consider to use `--install-types --non-interactive` as well, since for some package type annotations exist but have to be installed?